unstructured/partition/ocr.py

```diff
@@ -0,0 +1,481 @@
+                    )
+                    merged_page_layouts.append(merged_page_layout)
+                return DocumentLayout.from_pages(merged_page_layouts)
+        else:
+            with tempfile.TemporaryDirectory() as temp_dir:
+                _image_paths = pdf2image.convert_from_path(
+                    filename,
+                    dpi=pdf_image_dpi,
+                    output_folder=temp_dir,
+                    paths_only=True,
+                )
+                image_paths = cast(List[str], _image_paths)
+                for i, image_path in enumerate(image_paths):
+                    with PILImage.open(image_path) as image:
+                        merged_page_layout = supplement_page_layout_with_ocr(
+                            inferred_layout.pages[i],
+                            image,
+                            ocr_languages=ocr_languages,
+                            ocr_mode=ocr_mode,
+                        )
+                        merged_page_layouts.append(merged_page_layout)
+                return DocumentLayout.from_pages(merged_page_layouts)
+    except Exception as e:
+        if os.path.isdir(filename) or os.path.isfile(filename):
+            raise e
+        else:
+            raise FileNotFoundError(f'File "{filename}" not found!') from e
+
+
+def supplement_page_layout_with_ocr(
+    inferred_page_layout: "PageLayout",
+    image: PILImage,
+    ocr_languages: str = "eng",
+    ocr_mode: str = OCRMode.FULL_PAGE.value,
+) -> "PageLayout":
+    """
+    Supplement an inferred PageLayout with OCR results depending on OCR mode.
+    If mode is "entire_page", we get the OCR layout for the entire image and
+    merge it with inferred PageLayout.
+    If mode is "individual_blocks", we find the elements from inferred PageLayout
+    with no text and add text from OCR to each element.
+    """
+    entrie_page_ocr = os.getenv("ENTIRE_PAGE_OCR", "tesseract").lower()
+    # TODO(yuming): add tests for paddle with ENTIRE_PAGE_OCR env
+    # see CORE-1886
+    if entrie_page_ocr not in ["paddle", "tesseract"]:
+        raise ValueError(
+            "Environment variable ENTIRE_PAGE_OCR",
+            " must be set to 'tesseract' or 'paddle'.",
+        )
+    if ocr_mode == OCRMode.FULL_PAGE.value:
+        ocr_layout = get_ocr_layout_from_image(
+            image,
+            ocr_languages=ocr_languages,
+            entrie_page_ocr=entrie_page_ocr,
+        )
+        merged_page_layout_elements = merge_inferred_layout_with_ocr_layout(
+            inferred_page_layout.elements,
+            ocr_layout,
+        )
+        inferred_page_layout.elements[:] = merged_page_layout_elements
+        return inferred_page_layout
+    elif ocr_mode == OCRMode.INDIVIDUAL_BLOCKS.value:
+        elements = inferred_page_layout.elements
+        for i, element in enumerate(elements):
+            if element.text == "":
+                cropped_image = image.crop((element.x1, element.y1, element.x2, element.y2))
```

Comment:
You'll need to implement the same padding logic as in the `unstructured inference` repo - 
https://github.com/Unstructured-IO/unstructured-inference/blob/main/unstructured_inference/inference/elements.py#L270-L271

Reply:
thanks so much for catching this! this fixed a unit test after i merge from main that change default model to yolox